mbbsd/vote.c

```diff
@@ -519,6 +545,97 @@ vote_view_all(const char *bname)
 	return FULLUPDATE;
 }
 
+int
+vote_logconf_ask(vote_logconf_t *logconf) {
+    char inbuf[STRLEN];
+    logconf->log_id = 0;
+    logconf->log_ip = 0;
+    logconf->log_date = 0;
+    logconf->log_choice = 0;
+
+    getdata(14, 0,
+	    "�O�_�O��벼�b���G(y)�O��벼�b��[n]���O��:[N]",
+	    inbuf, 2, LCECHO);
+    if (inbuf[0] == 'y')
+	logconf->log_id = 1;
+
+    getdata(15, 0,
+	    "�O�_�O��벼IP�G(y)�O��벼IP[n]���O��:[N]",
+	    inbuf, 2, LCECHO);
+    if (inbuf[0] == 'y')
+	logconf->log_ip = 1;
+
+    getdata(16, 0,
+	    "�O�_�O��벼�ɶ��G(y)�O��벼�ɶ�[n]���O��:[N]",
+	    inbuf, 2, LCECHO);
+    if (inbuf[0] == 'y')
+	logconf->log_date = 1;
+
+    getdata(17, 0,
+	    "�O�_�O��벼�ﶵ�G(y)�O��벼�ﶵ[n]���O��:[N]",
+	    inbuf, 2, LCECHO);
+    if (inbuf[0] == 'y')
+	logconf->log_choice = 1;
+
+    return 0;
+}
+
+int
+vote_logconf_save(const vote_logconf_t *logconf, const char *fpath) {
+    FILE *fp = fopen(fpath, "w");
+    if (fp) {
+	fprintf(fp, "%d %d %d %d", logconf->log_id, logconf->log_ip, logconf->log_date, logconf->log_choice);
+	fclose(fp);
+	return 0;
+    } else 
+	return -1;
+}
+
+int
+vote_logconf_load(const char *fpath, vote_logconf_t *logconf) {
+    FILE *flogconf = fopen(fpath, "r");
+    if (flogconf) {
+	if (fscanf(flogconf, "%d%d%d%d", &(logconf->log_id), &(logconf->log_ip), &(logconf->log_date), &(logconf->log_choice)) != 4) {
+	    logconf->log_id = 0;
+	    logconf->log_ip = 0;
+	    logconf->log_date = 0;
+	    logconf->log_choice = 0;
+	}
+	fclose(flogconf);
+	return 0;
+    } else
+	return -1;
+}
+
+void
+print_vote_logconf_feature(int enabled, const char *feature) {
+    prints("     �����벼�N %s" ANSI_RESET "�O��ä��}�z�� %s%s" ANSI_RESET "\n",
+	   enabled ? ANSI_COLOR(1;31) "�|   " : ANSI_COLOR(1;37) "���| ",
+	   enabled ? ANSI_COLOR(1;31) : ANSI_COLOR(1;37), feature);
+}
+
+void
+vote_logconf_print(const vote_logconf_t *logconf) {
+    if (logconf->log_id && logconf->log_choice) {
+        move(15, 5);
+        prints("�ЯS�O�`�N " ANSI_COLOR(1;31) "�����벼���O�W�벼�A�z���b���P�벼�ﶵ�N�P�ɳQ�O��ä��}" ANSI_RESET);
+    }
+
+    move(18, 0);
+    print_vote_logconf_feature(logconf->log_id, "�b�� (ID)");
+
+    move(19, 0);
```

Comment:
You don't need `move` every time since `print_vote_logconf_feature` prints a new line. Please amend other `move` then `prints` as well.